abci/client/socket_client.go

```diff
@@ -210,12 +210,12 @@ func (cli *socketClient) didRecvResponse(res *types.Response) error {
 	// Get the first ReqRes.
 	next := cli.reqSent.Front()
 	if next == nil {
-		return fmt.Errorf("unexpected response %T when no call was made", res.Value)
+		return ErrUnexpectedResponse{Response: *res, Comment: "No call was made"}
 	}
 
 	reqres := next.Value.(*ReqRes)
 	if !resMatchesReq(reqres.Request, res) {
-		return fmt.Errorf("unexpected response %T to the request %T", res.Value, reqres.Request.Value)
+		return ErrUnexpectedResponse{Response: *res, Comment: fmt.Sprintf("Unexpected response to the request %T", reqres.Request.Value)}
```

Comment:
Why passing the whole `Response`,  rather than just `Value`?

Reply:
> 	Value isResponse_Value `protobuf_oneof:"value"`

the type of a Value is not exported by `types`. it is generated, so I personally do not think it is worth making it exported (at least if it is possible)